### PR TITLE
Add dataset period options and new anonymous metrics

### LINKED ISSUE #50: Record Glue ETL performance metrics
**Is your feature request related to a problem? Please describe.**

AWS Solutions measures solution usage through anonymous metrics in order to help guide people resources to where they can have the best impact. Currently this solution only reports stack launch metrics, but a better measure of usage would also include information about ETL workloads, so we can answer questions like, "How much data has this solution transformed for AMC?".

**Describe the feature you'd like**

Record anonymous metrics to measure the workloads processed by the Glue ETL job.

### PR DESCRIPTION
**Pull request by ianwow**
*Issue #, if available:*

Closes #50, #26

*Description of changes:*

Allow users to specify timeseries partition size.
Record anonymous metrics to measure the workloads processed by the Glue ETL job.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ianwow** [comment]
Here's an example of the metrics introduced by this PR:
```
METRICS SENT:: b'
{
    "Solution": "SO0222",
    "UUID": "d7e35815-d843-4683-9595-ccd9258116cf",
    "TimeStamp": "2023-01-18T23:46:58.247958",
    "Data": {
        "numBytes": 232168,
        "numRows": 3123,
        "datasetType": "FACT",
        "glueJobDuration": 73.682465
    }
}
```

File: source/tests/e2e/test_app.py [hunk 1]
```diff
@@ -57,5 +56,17 @@ def test_everything(browser, test_environment, stack_resources):
     rows = browser.find_elements("xpath", xpath)
     assert len(rows) > 0
 
+    # open Step 2
+    browser.find_element(By.ID, "step2").click()
+    wait.until(EC.presence_of_element_located((By.ID, "dataset_type_options")))
```
Comment (ikp4success) [inline source/tests/e2e/test_app.py:61]: NP: i wonder if a fixed amount of timeout should be set like [this](https://selenium-python.readthedocs.io/waits.html?highlight=wait.until#explicit-waits). This way it's not freezing the whole process, waiting for element to be available. Is also good to know why is taking long for element to show up, for good user experience.

**Review by ikp4success — CHANGES_REQUESTED** [review]

File: source/anonymous-data-logger/anonymous-data-logger.py [hunk 2]
```diff
@@ -31,18 +31,20 @@ def handler(event, context):
...
+            resource = event['ResourceProperties']['Resource']
+            config = event['ResourceProperties']
+            # Remove ServiceToken (lambda arn) to avoid sending AccountId
+            config.pop("ServiceToken", None)
+            config.pop("Resource", None)
+            # Add some useful fields related to stack change
+            config["CFTemplate"] = (
+                    request_type + "d"
+            )  # Created, Updated, or Deleted
+            response_data = {}
```
[the remark this answers is not in the document]
Comment (ianwow) [reply source/anonymous-data-logger/anonymous-data-logger.py:46]: Do you still think that after looking at this line?

https://github.com/aws-solutions/amazon-marketing-cloud-uploader-from-aws/blob/5128b04f158f323a7e091ef95f66249e8d795c68/source/anonymous-data-logger/anonymous-data-logger.py#L55

[the remark this answers is not in the document]
Comment (ikp4success) [reply source/anonymous-data-logger/anonymous-data-logger.py:60]: ```cfn.send(event, context, 'SUCCESS', response_data, response_data['UUID'])```
replace with this

File: source/tests/e2e/test_app.py (hunk 1 above)
Comment (ianwow) [reply source/tests/e2e/test_app.py:61]: I am using a fixed timeout. Not sure what you are suggesting here. Also automating the process of troubleshooting slow UX is outside the scope of this PR.

Comment (ikp4success) [reply source/tests/e2e/test_app.py:61]: ah ok, i see, please disregard.

File: source/api/app.py [hunk 3]
```diff
@@ -79,6 +79,10 @@ def create_dataset():
     log_request_parameters()
     try:
         body = app.current_request.json_body['body']
+        if body['period'] == 'autodetect':
+            # Initialize the dataset period to P1D. This will be updated later
+            # when the AWS Glue job measures the actual dataset period.
+            body['period'] = 'P1D'
```
[the remark this answers is not in the document]
Comment (ianwow) [reply source/api/app.py:85]: Your suggestion improves terseness at the expense of readability. How do you recommend we justify that tradeoff?

[the remark this answers is not in the document]
Comment (ikp4success) [reply source/api/app.py:85]: it was a suggestion.

File: source/glue/amc_transformations.py [hunk 4]
```diff
@@ -484,14 +519,15 @@ def state_transformations(text):
             df_partition2[timestamp_column] = df_partition2[timestamp_column].dt.strftime(datetime_format)
             df_partition = df_partition.append(df_partition2, ignore_index=True)
     # write the last timestamp to s3
-    output_file = 's3://'+output_bucket+'/'+amc_str+'/'+dataset_id+'/'+timeseries_partition_size+'/'+filename+'-'+timestamp_str_old+'.gz'
+    output_file = 's3://' + output_bucket + '/' + amc_str + '/' + dataset_id + '/' + timeseries_partition_size + '/' + filename + '-' + timestamp_str_old + '.gz'
```
[the remark this answers is not in the document]
Comment (ianwow) [reply source/glue/amc_transformations.py:522]: This should not block this PR. If you think we need to f-strings, please create an issue for that so we can prioritize it along with other feature / bug requests.

File: source/glue/amc_transformations.py [hunk 5]
```diff
@@ -513,3 +551,26 @@ def state_transformations(text):
         'output files': output_file
     }
     print(output)
+
+###############################
+# SAVE PERFORMANCE METRICS
+###############################
+
+if enable_anonymous_data == 'true':
```
[the remark this answers is not in the document]
Comment (ianwow) [reply source/glue/amc_transformations.py:559]: This should not block this PR. If you think this is necessary, please create an issue for that so we can prioritize it along with other feature / bug requests.

**Review by ikp4success — APPROVED** [review]

**Pull request merged**